### ISSUE TITLE
Return range in the lint result

### ISSUE THREAD

**Issue by sindresorhus** [eslint/eslint#3307]
Currently ESLint returns the `line` and `column` when linting. However, with editor plugins it would be useful to get the full range so instead of just visually showing where the violation starts it could mark the whole range of the violation.

For example, [here](https://github.com/AtomLinter/linter-eslint/blob/923ae69c1809ea85e0013daec266926ebb7aed77/lib/linter-eslint.coffee#L119) the Atomlinter ESLint plugin could use the range instead of just marking the first character.
##

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
:+1:

However, I guess it shows wrong ranges if semicolons are omitted.
Related: https://github.com/eslint/espree/issues/41

**Comment by ilyavolodin** [comment]
I think this would be great to add. Unfortunately, it requires manually modifying every single rule we have (or at least most of them).

**Comment by nzakas** [comment]
Yeah, this would be a significant change (need to update every rule, plus change the signature of `report()`) so I'm not sure if it's realistic to do this. Plus, third-party rules would all have to be updated if you assume every result will have a range.

@sindresorhus if only a handful of rules returned range information, would that be useful to you?

**Comment by ilyavolodin** [comment]
@nzakas We could probably modify it so that any time we use `context.report` and pass it an existing node, we extract both start and end position from the node. That would only affect some of the rules, but it's probably a good start.

**Comment by sindresorhus** [comment]
> Plus, third-party rules would all have to be updated if you assume every result will have a range.

You don't have to assume that though. How about adding a `range` property to rule results incrementally while still preserving `line`/`column`. That way tools can first check `range` and fall back to `line`/`column`. Nothing breaks for existing consumers and external rules.

> @sindresorhus if only a handful of rules returned range information, would that be useful to you?

Yes, even just a few rules supporting it would improve the UI for editor linters.

**Comment by nzakas** [comment]
OK, I'm not sure we can commit to this at this point. Right now, each message contains the exact same data, and it's hard to tell what effect having different data on different messages would have in the ecosystem.

**Comment by ilyavolodin** [comment]
I marked this as backlog, because we can't commit to this right now and put it on the roadmap, but I think ideally, this is something we want to do at some point.

**Comment by nzakas** [comment]
:+1:

**Comment by mysticatea** [comment]
I will work on this.

My plan is that I add `lineEnd` and `columnEnd` properties to the result of `eslint.verify()`.
And I will check all core rules.

```js
{
    fatal: false,
    severity: 2,
    ruleId: "semi",
    line: 1,
    column: 23,
    lineEnd: 1,
    columnEnd: 23,
    message: "Expected a semicolon.",
    fix: {
        range: [1, 15],
        text: ";"
    }
}
```

Those properties are calculated from the arguments of `context.report(params)` in the following way.

```js
var line, column, lineEnd, columnEnd;

var loc = params.loc || params.node.loc.start;
if (loc.end) {
    line = loc.start.line;
    column = loc.start.column;
    lineEnd = loc.end.line;
    columnEnd = loc.end.column;
} else {
    line = loc.line;
    column = loc.column;
}
```

For Example:

```js
// Pattern 1
context.report({
    node: node,
    loc: token.loc,       // lineEnd = token.loc.end.line;
    message: "test"
});

// Pattern 2
context.report({
    node: node,
    loc: token.loc.start, // lineEnd is undefined;
    message: "test"
});

// Pattern 3
context.report({
    node: node,           // lineEnd is undefined;
    message: "test"
});
```

In short, the current code of core rules and plugins don't make `lineEnd` and `columnEnd`.
We should update rules with using `loc` property as `{start: {line: number, column: number}, end: {line: number, column: number}}`.

This is in order to keep compatibility of editors.
Especially in the case of the pattern 3, if we use `node.loc.end`, editors would show unexpected many red lines.

@eslint/eslint-team I want to ask opinions.

**Comment by ilyavolodin** [comment]
```js
var loc = params.loc || params.node.loc.start;
```
This wouldn't work if only node is passed in, but I get the general idea. :+1: from me

**Comment by nzakas** [comment]
Sounds good to me, too.

**Comment by nzakas** [comment]
@mysticatea was lineEnd added?

**Comment by mysticatea** [comment]
The implement of this issue has done in #6640.
Tweaking of core rules remains.

**Comment by Arcanemagus** [comment]
Have any of the core rules been updated yet?

**Comment by mysticatea** [comment]
Yes, only `no-unreachable` rule uses this feature for now: #6583

**Comment by nzakas** [comment]
@mysticatea if you want to keep this issue open, can you make a list of rules to that need to be addressed?

**Comment by mysticatea** [comment]
I will start to work on this within a week.
(I re-checked rules, then I found that several rules are using the range feature.)

- [ ] [accessor-pairs](http://eslint.org/docs/rules/accessor-pairs)
- [ ] [array-bracket-spacing](http://eslint.org/docs/rules/array-bracket-spacing)
- [ ] [array-callback-return](http://eslint.org/docs/rules/array-callback-return)
- [ ] [arrow-body-style](http://eslint.org/docs/rules/arrow-body-style)
- [ ] [arrow-parens](http://eslint.org/docs/rules/arrow-parens)
- [ ] [arrow-spacing](http://eslint.org/docs/rules/arrow-spacing)
- [ ] [block-scoped-var](http://eslint.org/docs/rules/block-scoped-var)
- [ ] [block-spacing](http://eslint.org/docs/rules/block-spacing)
- [ ] [brace-style](http://eslint.org/docs/rules/brace-style)
- [ ] [callback-return](http://eslint.org/docs/rules/callback-return)
- [ ] [camelcase](http://eslint.org/docs/rules/camelcase)
- [ ] [class-methods-use-this](http://eslint.org/docs/rules/class-methods-use-this)
- [ ] [comma-dangle](http://eslint.org/docs/rules/comma-dangle)
- [ ] [comma-spacing](http://eslint.org/docs/rules/comma-spacing)
- [ ] [comma-style](http://eslint.org/docs/rules/comma-style)
- [ ] [complexity](http://eslint.org/docs/rules/complexity)
- [ ] [computed-property-spacing](http://eslint.org/docs/rules/computed-property-spacing)
- [ ] [consistent-return](http://eslint.org/docs/rules/consistent-return)
- [ ] [consistent-this](http://eslint.org/docs/rules/consistent-this)
- [ ] [constructor-super](http://eslint.org/docs/rules/constructor-super)
- [ ] [curly](http://eslint.org/docs/rules/curly)
- [ ] [default-case](http://eslint.org/docs/rules/default-case)
- [ ] [dot-location](http://eslint.org/docs/rules/dot-location)
- [ ] [dot-notation](http://eslint.org/docs/rules/dot-notation)
- [ ] [eol-last](http://eslint.org/docs/rules/eol-last)
- [ ] [eqeqeq](http://eslint.org/docs/rules/eqeqeq)
- [ ] [func-call-spacing](http://eslint.org/docs/rules/func-call-spacing)
- [ ] [func-name-matching](http://eslint.org/docs/rules/func-name-matching)
- [ ] [func-names](http://eslint.org/docs/rules/func-names)
- [ ] [func-style](http://eslint.org/docs/rules/func-style)
- [ ] [generator-star-spacing](http://eslint.org/docs/rules/generator-star-spacing)
- [ ] [global-require](http://eslint.org/docs/rules/global-require)
- [ ] [guard-for-in](http://eslint.org/docs/rules/guard-for-in)
- [ ] [handle-callback-err](http://eslint.org/docs/rules/handle-callback-err)
- [ ] [id-blacklist](http://eslint.org/docs/rules/id-blacklist)
- [ ] [id-length](http://eslint.org/docs/rules/id-length)
- [ ] [id-match](http://eslint.org/docs/rules/id-match)
- [ ] [indent](http://eslint.org/docs/rules/indent)
- [ ] [init-declarations](http://eslint.org/docs/rules/init-declarations)
- [ ] [jsx-quotes](http://eslint.org/docs/rules/jsx-quotes)
- [ ] [key-spacing](http://eslint.org/docs/rules/key-spacing)
- [ ] [keyword-spacing](http://eslint.org/docs/rules/keyword-spacing)
- [ ] [line-comment-position](http://eslint.org/docs/rules/line-comment-position)
- [ ] [linebreak-style](http://eslint.org/docs/rules/linebreak-style)
- [ ] [lines-around-comment](http://eslint.org/docs/rules/lines-around-comment)
- [ ] [lines-around-directive](http://eslint.org/docs/rules/lines-around-directive)
- [ ] [max-depth](http://eslint.org/docs/rules/max-depth)
- [ ] [max-len](http://eslint.org/docs/rules/max-len)
- [ ] [max-lines](http://eslint.org/docs/rules/max-lines)
- [ ] [max-nested-callbacks](http://eslint.org/docs/rules/max-nested-callbacks)
- [ ] [max-params](http://eslint.org/docs/rules/max-params)
- [ ] [max-statements-per-line](http://eslint.org/docs/rules/max-statements-per-line)
- [ ] [max-statements](http://eslint.org/docs/rules/max-statements)
- [ ] [multiline-ternary](http://eslint.org/docs/rules/multiline-ternary)
- [ ] [new-cap](http://eslint.org/docs/rules/new-cap)
- [ ] [new-parens](http://eslint.org/docs/rules/new-parens)
- [ ] [newline-after-var](http://eslint.org/docs/rules/newline-after-var)
- [ ] [newline-before-return](http://eslint.org/docs/rules/newline-before-return)
- [ ] [newline-per-chained-call](http://eslint.org/docs/rules/newline-per-chained-call)
- [ ] [no-alert](http://eslint.org/docs/rules/no-alert)
- [ ] [no-array-constructor](http://eslint.org/docs/rules/no-array-constructor)
- [ ] [no-bitwise](http://eslint.org/docs/rules/no-bitwise)
- [ ] [no-caller](http://eslint.org/docs/rules/no-caller)
- [ ] [no-case-declarations](http://eslint.org/docs/rules/no-case-declarations)
- [ ] [no-catch-shadow](http://eslint.org/docs/rules/no-catch-shadow)
- [ ] [no-class-assign](http://eslint.org/docs/rules/no-class-assign)
- [ ] [no-cond-assign](http://eslint.org/docs/rules/no-cond-assign)
- [ ] [no-confusing-arrow](http://eslint.org/docs/rules/no-confusing-arrow)
- [x] [no-console](http://eslint.org/docs/rules/no-console)
- [ ] [no-const-assign](http://eslint.org/docs/rules/no-const-assign)
- [ ] [no-constant-condition](http://eslint.org/docs/rules/no-constant-condition)
- [ ] [no-continue](http://eslint.org/docs/rules/no-continue)
- [ ] [no-control-regex](http://eslint.org/docs/rules/no-control-regex)
- [ ] [no-debugger](http://eslint.org/docs/rules/no-debugger)
- [ ] [no-delete-var](http://eslint.org/docs/rules/no-delete-var)
- [ ] [no-div-regex](http://eslint.org/docs/rules/no-div-regex)
- [ ] [no-dupe-args](http://eslint.org/docs/rules/no-dupe-args)
- [ ] [no-dupe-class-members](http://eslint.org/docs/rules/no-dupe-class-members)
- [x] [no-dupe-keys](http://eslint.org/docs/rules/no-dupe-keys)
- [ ] [no-duplicate-case](http://eslint.org/docs/rules/no-duplicate-case)
- [ ] [no-duplicate-imports](http://eslint.org/docs/rules/no-duplicate-imports)
- [ ] [no-else-return](http://eslint.org/docs/rules/no-else-return)
- [ ] [no-empty-character-class](http://eslint.org/docs/rules/no-empty-character-class)
- [ ] [no-empty-function](http://eslint.org/docs/rules/no-empty-function)
- [ ] [no-empty-pattern](http://eslint.org/docs/rules/no-empty-pattern)
- [ ] [no-empty](http://eslint.org/docs/rules/no-empty)
- [ ] [no-eq-null](http://eslint.org/docs/rules/no-eq-null)
- [ ] [no-eval](http://eslint.org/docs/rules/no-eval)
- [ ] [no-ex-assign](http://eslint.org/docs/rules/no-ex-assign)
- [ ] [no-extend-native](http://eslint.org/docs/rules/no-extend-native)
- [ ] [no-extra-bind](http://eslint.org/docs/rules/no-extra-bind)
- [ ] [no-extra-boolean-cast](http://eslint.org/docs/rules/no-extra-boolean-cast)
- [ ] [no-extra-label](http://eslint.org/docs/rules/no-extra-label)
- [ ] [no-extra-parens](http://eslint.org/docs/rules/no-extra-parens)
- [ ] [no-extra-semi](http://eslint.org/docs/rules/no-extra-semi)
- [ ] [no-fallthrough](http://eslint.org/docs/rules/no-fallthrough)
- [ ] [no-floating-decimal](http://eslint.org/docs/rules/no-floating-decimal)
- [ ] [no-func-assign](http://eslint.org/docs/rules/no-func-assign)
- [ ] [no-global-assign](http://eslint.org/docs/rules/no-global-assign)
- [ ] [no-implicit-coercion](http://eslint.org/docs/rules/no-implicit-coercion)
- [ ] [no-implicit-globals](http://eslint.org/docs/rules/no-implicit-globals)
- [ ] [no-implied-eval](http://eslint.org/docs/rules/no-implied-eval)
- [ ] [no-inline-comments](http://eslint.org/docs/rules/no-inline-comments)
- [ ] [no-inner-declarations](http://eslint.org/docs/rules/no-inner-declarations)
- [ ] [no-invalid-regexp](http://eslint.org/docs/rules/no-invalid-regexp)
- [ ] [no-invalid-this](http://eslint.org/docs/rules/no-invalid-this)
- [ ] [no-irregular-whitespace](http://eslint.org/docs/rules/no-irregular-whitespace)
- [ ] [no-iterator](http://eslint.org/docs/rules/no-iterator)
- [ ] [no-label-var](http://eslint.org/docs/rules/no-label-var)
- [ ] [no-labels](http://eslint.org/docs/rules/no-labels)
- [ ] [no-lone-blocks](http://eslint.org/docs/rules/no-lone-blocks)
- [ ] [no-lonely-if](http://eslint.org/docs/rules/no-lonely-if)
- [ ] [no-loop-func](http://eslint.org/docs/rules/no-loop-func)
- [ ] [no-magic-numbers](http://eslint.org/docs/rules/no-magic-numbers)
- [ ] [no-mixed-operators](http://eslint.org/docs/rules/no-mixed-operators)
- [ ] [no-mixed-requires](http://eslint.org/docs/rules/no-mixed-requires)
- [ ] [no-mixed-spaces-and-tabs](http://eslint.org/docs/rules/no-mixed-spaces-and-tabs)
- [ ] [no-multi-spaces](http://eslint.org/docs/rules/no-multi-spaces)
- [ ] [no-multi-str](http://eslint.org/docs/rules/no-multi-str)
- [ ] [no-multiple-empty-lines](http://eslint.org/docs/rules/no-multiple-empty-lines)
- [ ] [no-native-reassign](http://eslint.org/docs/rules/no-native-reassign)
- [ ] [no-negated-condition](http://eslint.org/docs/rules/no-negated-condition)
- [ ] [no-negated-in-lhs](http://eslint.org/docs/rules/no-negated-in-lhs)
- [ ] [no-nested-ternary](http://eslint.org/docs/rules/no-nested-ternary)
- [ ] [no-new-func](http://eslint.org/docs/rules/no-new-func)
- [ ] [no-new-object](http://eslint.org/docs/rules/no-new-object)
- [ ] [no-new-require](http://eslint.org/docs/rules/no-new-require)
- [ ] [no-new-symbol](http://eslint.org/docs/rules/no-new-symbol)
- [ ] [no-new-wrappers](http://eslint.org/docs/rules/no-new-wrappers)
- [ ] [no-new](http://eslint.org/docs/rules/no-new)
- [ ] [no-obj-calls](http://eslint.org/docs/rules/no-obj-calls)
- [ ] [no-octal-escape](http://eslint.org/docs/rules/no-octal-escape)
- [ ] [no-octal](http://eslint.org/docs/rules/no-octal)
- [ ] [no-param-reassign](http://eslint.org/docs/rules/no-param-reassign)
- [ ] [no-path-concat](http://eslint.org/docs/rules/no-path-concat)
- [ ] [no-plusplus](http://eslint.org/docs/rules/no-plusplus)
- [ ] [no-process-env](http://eslint.org/docs/rules/no-process-env)
- [ ] [no-process-exit](http://eslint.org/docs/rules/no-process-exit)
- [ ] [no-proto](http://eslint.org/docs/rules/no-proto)
- [ ] [no-prototype-builtins](http://eslint.org/docs/rules/no-prototype-builtins)
- [ ] [no-redeclare](http://eslint.org/docs/rules/no-redeclare)
- [ ] [no-regex-spaces](http://eslint.org/docs/rules/no-regex-spaces)
- [ ] [no-restricted-globals](http://eslint.org/docs/rules/no-restricted-globals)
- [ ] [no-restricted-imports](http://eslint.org/docs/rules/no-restricted-imports)
- [ ] [no-restricted-modules](http://eslint.org/docs/rules/no-restricted-modules)
- [ ] [no-restricted-properties](http://eslint.org/docs/rules/no-restricted-properties)
- [ ] [no-restricted-syntax](http://eslint.org/docs/rules/no-restricted-syntax)
- [ ] [no-return-assign](http://eslint.org/docs/rules/no-return-assign)
- [ ] [no-script-url](http://eslint.org/docs/rules/no-script-url)
- [ ] [no-self-assign](http://eslint.org/docs/rules/no-self-assign)
- [ ] [no-self-compare](http://eslint.org/docs/rules/no-self-compare)
- [ ] [no-sequences](http://eslint.org/docs/rules/no-sequences)
- [ ] [no-shadow-restricted-names](http://eslint.org/docs/rules/no-shadow-restricted-names)
- [ ] [no-shadow](http://eslint.org/docs/rules/no-shadow)
- [ ] [no-spaced-func](http://eslint.org/docs/rules/no-spaced-func)
- [ ] [no-sparse-arrays](http://eslint.org/docs/rules/no-sparse-arrays)
- [ ] [no-sync](http://eslint.org/docs/rules/no-sync)
- [ ] [no-tabs](http://eslint.org/docs/rules/no-tabs)
- [ ] [no-template-curly-in-string](http://eslint.org/docs/rules/no-template-curly-in-string)
- [ ] [no-ternary](http://eslint.org/docs/rules/no-ternary)
- [ ] [no-this-before-super](http://eslint.org/docs/rules/no-this-before-super)
- [ ] [no-throw-literal](http://eslint.org/docs/rules/no-throw-literal)
- [ ] [no-trailing-spaces](http://eslint.org/docs/rules/no-trailing-spaces)
- [ ] [no-undef-init](http://eslint.org/docs/rules/no-undef-init)
- [ ] [no-undef](http://eslint.org/docs/rules/no-undef)
- [ ] [no-undefined](http://eslint.org/docs/rules/no-undefined)
- [ ] [no-underscore-dangle](http://eslint.org/docs/rules/no-underscore-dangle)
- [ ] [no-unexpected-multiline](http://eslint.org/docs/rules/no-unexpected-multiline)
- [ ] [no-unmodified-loop-condition](http://eslint.org/docs/rules/no-unmodified-loop-condition)
- [ ] [no-unneeded-ternary](http://eslint.org/docs/rules/no-unneeded-ternary)
- [x] [no-unreachable](http://eslint.org/docs/rules/no-unreachable)
- [ ] [no-unsafe-finally](http://eslint.org/docs/rules/no-unsafe-finally)
- [x] [no-unsafe-negation](http://eslint.org/docs/rules/no-unsafe-negation)
- [ ] [no-unused-expressions](http://eslint.org/docs/rules/no-unused-expressions)
- [ ] [no-unused-labels](http://eslint.org/docs/rules/no-unused-labels)
- [ ] [no-unused-vars](http://eslint.org/docs/rules/no-unused-vars)
- [ ] [no-use-before-define](http://eslint.org/docs/rules/no-use-before-define)
- [ ] [no-useless-call](http://eslint.org/docs/rules/no-useless-call)
- [ ] [no-useless-computed-key](http://eslint.org/docs/rules/no-useless-computed-key)
- [ ] [no-useless-concat](http://eslint.org/docs/rules/no-useless-concat)
- [ ] [no-useless-constructor](http://eslint.org/docs/rules/no-useless-constructor)
- [ ] [no-useless-escape](http://eslint.org/docs/rules/no-useless-escape)
- [ ] [no-useless-rename](http://eslint.org/docs/rules/no-useless-rename)
- [ ] [no-var](http://eslint.org/docs/rules/no-var)
- [ ] [no-void](http://eslint.org/docs/rules/no-void)
- [ ] [no-warning-comments](http://eslint.org/docs/rules/no-warning-comments)
- [ ] [no-whitespace-before-property](http://eslint.org/docs/rules/no-whitespace-before-property)
- [ ] [no-with](http://eslint.org/docs/rules/no-with)
- [ ] [object-curly-newline](http://eslint.org/docs/rules/object-curly-newline)
- [ ] [object-curly-spacing](http://eslint.org/docs/rules/object-curly-spacing)
- [ ] [object-property-newline](http://eslint.org/docs/rules/object-property-newline)
- [ ] [object-shorthand](http://eslint.org/docs/rules/object-shorthand)
- [ ] [one-var-declaration-per-line](http://eslint.org/docs/rules/one-var-declaration-per-line)
- [ ] [one-var](http://eslint.org/docs/rules/one-var)
- [ ] [operator-assignment](http://eslint.org/docs/rules/operator-assignment)
- [ ] [operator-linebreak](http://eslint.org/docs/rules/operator-linebreak)
- [ ] [padded-blocks](http://eslint.org/docs/rules/padded-blocks)
- [ ] [prefer-arrow-callback](http://eslint.org/docs/rules/prefer-arrow-callback)
- [ ] [prefer-const](http://eslint.org/docs/rules/prefer-const)
- [ ] [prefer-numeric-literals](http://eslint.org/docs/rules/prefer-numeric-literals)
- [ ] [prefer-reflect](http://eslint.org/docs/rules/prefer-reflect)
- [x] [prefer-rest-params](http://eslint.org/docs/rules/prefer-rest-params)
- [ ] [prefer-spread](http://eslint.org/docs/rules/prefer-spread)
- [ ] [prefer-template](http://eslint.org/docs/rules/prefer-template)
- [ ] [quote-props](http://eslint.org/docs/rules/quote-props)
- [ ] [quotes](http://eslint.org/docs/rules/quotes)
- [ ] [radix](http://eslint.org/docs/rules/radix)
- [ ] [require-jsdoc](http://eslint.org/docs/rules/require-jsdoc)
- [ ] [require-yield](http://eslint.org/docs/rules/require-yield)
- [ ] [rest-spread-spacing](http://eslint.org/docs/rules/rest-spread-spacing)
- [ ] [semi-spacing](http://eslint.org/docs/rules/semi-spacing)
- [ ] [semi](http://eslint.org/docs/rules/semi)
- [ ] [sort-imports](http://eslint.org/docs/rules/sort-imports)
- [x] [sort-keys](http://eslint.org/docs/rules/sort-keys)
- [ ] [sort-vars](http://eslint.org/docs/rules/sort-vars)
- [ ] [space-before-blocks](http://eslint.org/docs/rules/space-before-blocks)
- [ ] [space-before-function-paren](http://eslint.org/docs/rules/space-before-function-paren)
- [ ] [space-in-parens](http://eslint.org/docs/rules/space-in-parens)
- [ ] [space-infix-ops](http://eslint.org/docs/rules/space-infix-ops)
- [ ] [space-unary-ops](http://eslint.org/docs/rules/space-unary-ops)
- [ ] [spaced-comment](http://eslint.org/docs/rules/spaced-comment)
- [ ] [strict](http://eslint.org/docs/rules/strict)
- [ ] [symbol-description](http://eslint.org/docs/rules/symbol-description)
- [ ] [template-curly-spacing](http://eslint.org/docs/rules/template-curly-spacing)
- [ ] [unicode-bom](http://eslint.org/docs/rules/unicode-bom)
- [ ] [use-isnan](http://eslint.org/docs/rules/use-isnan)
- [ ] [valid-jsdoc](http://eslint.org/docs/rules/valid-jsdoc)
- [ ] [valid-typeof](http://eslint.org/docs/rules/valid-typeof)
- [ ] [vars-on-top](http://eslint.org/docs/rules/vars-on-top)
- [ ] [wrap-iife](http://eslint.org/docs/rules/wrap-iife)
- [ ] [wrap-regex](http://eslint.org/docs/rules/wrap-regex)
- [ ] [yield-star-spacing](http://eslint.org/docs/rules/yield-star-spacing)
- [ ] [yoda](http://eslint.org/docs/rules/yoda)

**Comment by kaicataldo** [comment]
Is this still being worked on?

**Comment by mysticatea** [comment]
Yes, I will do. I'm sorry, I'm busy for now. I think I can get time within the next month.

**Comment by not-an-aardvark** [comment]
I think this is basically fixed now that https://github.com/eslint/eslint/commit/b0c63f0a928cceff7cf6f060262dd7ac4bc0f8dc has been merged, so I'm closing this issue.
